calculatorUICOOP/ViewModels/MainPageViewModel.cs

```diff
@@ -54,116 +66,135 @@ public string DisplayContent
         public MainPageViewModel()
         {
             DisplayContent = "0";
-            NumberInputCommand = new Command<string>(ShowNumberOnDisplay);
-            ClearInputCommand = new Command(ClearScreen);
-            DeleteInputCommand = new Command(DeleteLastInput);
-            DivideInputCommand = new Command<string>(ShowDivideOnDisplay);
-            MultiplyInputCommand = new Command<string>(ShowMultiplyOnDisplay);
+            Input = "0";
+            NumberInputCommand = new Command<string>(AppendDigit);
+            DecimalInputCommand = new Command<string>(AppendDecimal);
             PlusInputCommand = new Command<string>(ShowPlusOnDisplay);
             MinusInputCommand = new Command<string>(ShowMinusOnDisplay);
+            MultiplyInputCommand = new Command<string>(ShowMultiplyOnDisplay);
+            DivideInputCommand = new Command<string>(ShowDivideOnDisplay);
             PercentInputCommand = new Command<string>(ShowPercentageOnDisplay);
-            DecimalInputCommand = new Command<string>(ShowDecimalOnDisplay);
             EqualsInputCommand = new Command(Equals);
+            ClearInputCommand = new Command(ClearScreen);
+            DeleteInputCommand = new Command(DeleteLastInput);
         }
         #endregion
 
         #region Methods
-        private void OnPropertyChanged([CallerMemberName] string text = null)
+        public void AppendDigit(string number)
```

Comment:
IMHO receiving the digit as a `char` instead of a `string` could prevent possibly mistakes and make more evident the purpose of this function. What do you think?

Reply:
The incoming value will be a string since it is the button's Text property. In this case, it will always be a single character in the form of a string, which is fine unless something other than 0-9 is bound to this command.

Reply:
Ok, I thought that, but now I tested it and for some reason it doesn't work.

Reply:
As the CommandParameter stores an object, according to [docs](https://docs.microsoft.com/pt-br/dotnet/api/xamarin.forms.button.commandparameter?view=xamarin-forms), why does that doesn't work?